APIManager.js

```diff
@@ -0,0 +1,119 @@
+class APIManager {
+    constructor() {
+        this.data = {
+            user: {
```

Comment:
you do not need to have all these keys starting in your `this.data` object. makes your code look a bit busy. You can delete these keys and it should still work